test/uk/gov/hmrc/economiccrimelevyregistration/services/IntegrationFrameworkServiceSpec.scala

```diff
@@ -0,0 +1,179 @@
+        }
+
+        result.getMessage shouldBe s"Subscription status $subscriptionStatus returned with unexpected idType $idType and value $idValue"
+    }
+
+    "throw an IllegalStateException when the subscription status is successful but there is no id type or value" in forAll {
+      (
+        businessPartnerId: String,
+        channel: Option[Channel]
+      ) =>
+        val subscriptionStatusResponse = SubscriptionStatusResponse(
+          subscriptionStatus = Successful,
+          idType = None,
+          idValue = None,
+          channel = channel
+        )
+
+        when(mockIntegrationFrameworkConnector.getSubscriptionStatus(ArgumentMatchers.eq(businessPartnerId))(any()))
+          .thenReturn(Future.successful(subscriptionStatusResponse))
+
+        val result = intercept[IllegalStateException] {
+          await(service.getSubscriptionStatus(businessPartnerId))
+        }
+
+        result.getMessage shouldBe s"Subscription status is ${subscriptionStatusResponse.subscriptionStatus} but there is no id type or value"
+    }
+
+    "throw an IllegalStateException when the subscription status is successful but there is an id type without an id value" in forAll {
+      (
+        businessPartnerId: String,
+        channel: Option[Channel]
+      ) =>
+        val subscriptionStatusResponse = SubscriptionStatusResponse(
+          subscriptionStatus = Successful,
+          idType = Some("ZECL"),
+          idValue = None,
+          channel = channel
+        )
+
+        when(mockIntegrationFrameworkConnector.getSubscriptionStatus(ArgumentMatchers.eq(businessPartnerId))(any()))
+          .thenReturn(Future.successful(subscriptionStatusResponse))
+
+        val result = intercept[IllegalStateException] {
+          await(service.getSubscriptionStatus(businessPartnerId))
+        }
+
+        result.getMessage shouldBe s"Subscription status is ${subscriptionStatusResponse.subscriptionStatus} but there is no id type or value"
+    }
+
+    "throw an IllegalStateException when the subscription status is successful but there is an id value without an id type" in forAll {
+      (
+        businessPartnerId: String,
+        idValue: String,
+        channel: Option[Channel]
+      ) =>
+        val subscriptionStatusResponse = SubscriptionStatusResponse(
+          subscriptionStatus = Successful,
+          idType = None,
+          idValue = Some(idValue),
+          channel = channel
+        )
+
+        when(mockIntegrationFrameworkConnector.getSubscriptionStatus(ArgumentMatchers.eq(businessPartnerId))(any()))
+          .thenReturn(Future.successful(subscriptionStatusResponse))
+
+        val result = intercept[IllegalStateException] {
+          await(service.getSubscriptionStatus(businessPartnerId))
+        }
+
+        result.getMessage shouldBe s"Subscription status is ${subscriptionStatusResponse.subscriptionStatus} but there is no id type or value"
+    }
+  }
```

Comment:
Think we can condense these 3 tests down in to a single test and the expected outcome is the same in each case. It's just the input that differs, you could have a Table with the 3 different inputs in a single column of type `(Option[String], Option[String])` (idType and idValue)

Reply:
👍